include/boost/thread/shared_mutex.hpp

```diff
@@ -13,13 +13,20 @@
 #include <boost/thread/detail/config.hpp>
 #if defined(BOOST_THREAD_PLATFORM_WIN32)
 #if defined(BOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN)
+#if defined(BOOST_THREAD_V2_SHARED_MUTEX)
+#include <boost/thread/v2/shared_mutex.hpp>
```

Comment:
I don't remember why I've two generic implementation. Wondering what are the differences?

Reply:
The v2/shared_mutex.hpp implementation was one you added back in 2013 based on an implementation by H. Hinnant. I don't know if it's a better implementation than the existing one or not.